src/sparse/KokkosSparse_spgemm_jacobi.hpp

```diff
@@ -0,0 +1,249 @@
+		     typename cscalar_nnz_view_t_::non_const_value_type>::value,
+		     "KokkosSparse::spgemm_jacobi: Output matrix scalar view must be non-const.");
+
+      static_assert (std::is_same<typename KernelHandle::const_size_type,
+		     typename alno_row_view_t_::const_value_type>::value,
+		     "KokkosSparse::spgemm_jacobi: Size type of left handside matrix should be same as kernelHandle sizetype.");
+
+      static_assert (std::is_same<typename KernelHandle::const_size_type,
+		     typename blno_row_view_t_::const_value_type>::value,
+		     "KokkosSparse::spgemm_jacobi: Size type of right handside matrix should be same as kernelHandle sizetype.");
+
+      static_assert (std::is_same<typename KernelHandle::const_size_type,
+		     typename clno_row_view_t_::const_value_type>::value,
+		     "KokkosSparse::spgemm_jacobi: Size type of output matrix should be same as kernelHandle sizetype.");
+
+      static_assert (std::is_same<typename KernelHandle::const_nnz_lno_t,
+		     typename alno_nnz_view_t_::const_value_type>::value,
+		     "KokkosSparse::spgemm_jacobi: lno type of left handside matrix should be same as kernelHandle lno_t.");
+
+      static_assert (std::is_same<typename KernelHandle::const_nnz_lno_t,
+		     typename blno_nnz_view_t_::const_value_type>::value,
+		     "KokkosSparse::spgemm_jacobi: lno type of right handside matrix should be same as kernelHandle lno_t.");
+
+      static_assert (std::is_same<typename KernelHandle::const_nnz_lno_t,
+		     typename clno_nnz_view_t_::const_value_type>::value,
+		     "KokkosSparse::spgemm_jacobi: lno type of output matrix should be same as kernelHandle lno_t.");
+
+      static_assert (std::is_same<typename KernelHandle::const_nnz_scalar_t,
+		     typename ascalar_nnz_view_t_::const_value_type>::value,
+		     "KokkosSparse::spgemm_jacobi: scalar type of left handside matrix should be same as kernelHandle scalar.");
+
+      static_assert (std::is_same<typename KernelHandle::const_nnz_scalar_t,
+		     typename bscalar_nnz_view_t_::const_value_type>::value,
+		     "KokkosSparse::spgemm_jacobi: scalar type of right handside matrix should be same as kernelHandle scalar.");
+
+      static_assert (std::is_same<typename KernelHandle::const_nnz_scalar_t,
+		     typename cscalar_nnz_view_t_::const_value_type>::value,
+		     "KokkosSparse::spgemm_jacobi: scalar type of output matrix should be same as kernelHandle scalar.");
+
+
+      if (transposeA || transposeB){
+	throw std::runtime_error ("SpGEMM is not implemented for Transposes yet. "
+				  "If you need this case please let kokkos-kernels developers know.\n");
+      }
+
+
+      typedef typename KernelHandle::const_size_type c_size_t;
+      typedef typename KernelHandle::const_nnz_lno_t c_lno_t;
+      typedef typename KernelHandle::const_nnz_scalar_t c_scalar_t;
+
+      typedef typename KernelHandle::HandleExecSpace c_exec_t;
+      typedef typename KernelHandle::HandleTempMemorySpace c_temp_t;
```

Comment:
Do we actually use TempMemorySpace and PersistentMemorySpace ?

Reply:
I am not able to tell if this is absolutely required or not. TempMemorySpace and PersistentMemorySpace go into the const_handle_type, which is used in the KokkosSparse::Impl::SPGEMM_JACOBI call at the end of this file.

Reply:
I mean, this is a copy of the file KokkosSparse_spgemm_numeric.hpp which is updated for jacobi accordingly.

Reply:
Understood, this requires a cleanup in spgemm as well. I think the handle might have reasonable defaults. If you think it is hard to remove we can come to it w/ clean up pass for both codes later.

Reply:
Yes, it is hard to remove at the moment. Let's do it later.